### PR TITLE
fix(tab-bar): fix the activation of tab without previous active tab

### LINKED ISSUE #4226: [TabBar] MDCTabBar: At least one tab must have mdc-tab--active class
There must be one tab element in the tab bar component with a class of mdc-tab--active or an error is thrown in the console: Cannot read property 'deactivate' of undefined when switching tabs

### What MDC Web Version are you using?

Latest from master branch.

### What browser(s) is this bug affecting?

All

### What are the steps to reproduce the bug?

See this [CodePen](https://codepen.io/robjacobs/pen/Kboapr?editors=1011) example.  Notice the error in the console that results from trying to use the tab bar activateTab method.  Notice the error is also thrown when clicking on tabs.  Uncommenting the last tab in the example fixes the issue as it has a mdc-tab--active class.

### What is the expected behavior?

Using the mdc-tab--active class to initially set the active tab should not be required.  You should be able to use the mdc-tab activateTab method to set the initial active tab.

### What is the actual behavior?

You must use the mdc-tab--active on at least one child mdc-tab element.

### PR DESCRIPTION
**Pull request by stof**
If there is no previous active tab, the index will be `-1`, which will be trigger errors in the adapter of the component when trying to call methods on `this.tabList_[index]`. This is now handled gracefully for the case when there is no previously active tab.

Fixes #4226

### CODE REVIEW

File: packages/mdc-tab-bar/adapter.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export interface MDCTabBarAdapter {
    * @param index The index of the tab to activate
    * @param clientRect The client rect of the previously active Tab Indicator
    */
-  activateTabAtIndex(index: number, clientRect: ClientRect): void;
+  activateTabAtIndex(index: number, clientRect?: ClientRect): void;
```
Comment (stof) [inline packages/mdc-tab-bar/adapter.ts]: the implementation in the component adapter defers to a MDCTab method which already uses an optional parameter. That's why it does not involve other changes.

File: packages/mdc-tab-bar/foundation.ts [hunk 2]
```diff
@@ -97,8 +97,13 @@ export class MDCTabBarFoundation extends MDCFoundation<MDCTabBarAdapter> {
       return;
     }
 
-    this.adapter_.deactivateTabAtIndex(previousActiveIndex);
-    this.adapter_.activateTabAtIndex(index, this.adapter_.getTabIndicatorClientRectAtIndex(previousActiveIndex));
+    let previousClientRect;
+    if (-1 !== previousActiveIndex) {
```
Comment (patrickrodee) [inline packages/mdc-tab-bar/foundation.ts]: Let's reverse this to `if (previousActiveIndex !== -1)`

File: packages/mdc-tab-bar/adapter.ts (hunk 1 above)
Comment (patrickrodee) [reply packages/mdc-tab-bar/adapter.ts]: Nice!

**Comment by stof** [comment]
@patrickrodee done. For the test, I added a new test covering that behavior instead of changing the one you linked, as it was covering a different thing.

**Pull request merged**